### PR TITLE
Enable flexible tenant login

### PR DESCRIPTION
**Pull request by stevenhurwitt**
-Changed **/KLS/client/src/extension.ts** command *kuskus.loadSymbols* to include an authority/tenant id parameter instead of using the default "common". Also modified the change state listener to throw errors regarding this parameter.

-Changed **/KLS/server/src/kustoConnection.ts** function *getClient* to take authId parameter instead of "common" string.

-Changed **/KLS/server/src/server.ts** request kuskus.loadSymbols to include authId in request. **did not** change the notifications here out of laziness. sorry.

### CODE REVIEW

File: kusto-language-server/client/src/extension.ts [hunk 1]
```diff
@@ -105,6 +105,16 @@ commands.registerCommand('kuskus.loadSymbols', async () => {
 			return;
 		}
 
+		const authId = await window.showInputBox({
+			ignoreFocusOut: true,
+			placeHolder: 'common',
```
Comment (rosshamish) [inline kusto-language-server/client/src/extension.ts:110]: What are some other common values for authority / tenant id? I'm not familiar with it. I suppose my use case is the "common" one.

File: kusto-language-server/client/src/extension.ts [hunk 2]
```diff
@@ -105,6 +105,16 @@ commands.registerCommand('kuskus.loadSymbols', async () => {
...
+			prompt: 'Authority/Tenant ID'
+		});
+		if (!authId) {
```
Comment (rosshamish) [inline kusto-language-server/client/src/extension.ts:113]: If the user just immediately hits enter, does `window.showInputBox` return the placeholder value of "common" or will it return null / empty string? Aka if the user just hits enter, do they get this error message or nah?

File: kusto-language-server/client/src/extension.ts [hunk 3]
```diff
@@ -105,6 +105,16 @@ commands.registerCommand('kuskus.loadSymbols', async () => {
 			return;
 		}
 
+		const authId = await window.showInputBox({
+			ignoreFocusOut: true,
+			placeHolder: 'common',
+			prompt: 'Authority/Tenant ID'
```
Comment (rosshamish) [inline kusto-language-server/client/src/extension.ts:111]: Is there some way to provide a bit more info to the user on what "authority/tenant ID" means and how they'd find out what they should use? E.g. is there another param to `.showInputBox` like `hint` or similar? I wouldn't know what to do here

File: kusto-language-server/client/src/extension.ts (hunk 1 above)
Comment (stevenhurwitt) [reply kusto-language-server/client/src/extension.ts:110]: I believe they're specific to your azure AD, for example if i'm using Company A resources, then i would use their tenant/authority id XXXX-XXXX-XXXX and for Company B I would use their tenant id YYYY-YYYY-YYYYY.

File: kusto-language-server/client/src/extension.ts (hunk 2 above)
Comment (stevenhurwitt) [reply kusto-language-server/client/src/extension.ts:113]: i'll test this and try to enable that functionality if it doesn't do that.

File: kusto-language-server/client/src/extension.ts (hunk 3 above)
Comment (stevenhurwitt) [reply kusto-language-server/client/src/extension.ts:111]: The tenant id is available in the Azure Portal on Azure Active Directory home page. I'll do some brainstorming to see how to best convey this info to potential users.

**Comment by stevenhurwitt** [comment]
ok i pushed some slight changes to my current master - do i need to open another pull request?

basically i just made it so common is the value in tenant id, also just added a little info to the prompt that tenant id can be found in Azure Portal / Azure Active Directory

File: kusto-language-server/client/src/extension.ts (hunk 3 above)
Comment (stevenhurwitt) [reply kusto-language-server/client/src/extension.ts]: functionality i would really like to add would be something like in the [vscode-cosmosdb](https://github.com/Microsoft/vscode-cosmosdb) extension. If you scroll down on the readme, Will Lorey would be the readable name of the tenant id number and then you can see a list of databases, expand to get a list of tables, etc...

Comment (rosshamish) [reply kusto-language-server/client/src/extension.ts]: Nice yeah I think that would be a good future contribution.

File: kusto-language-server/package.json [hunk 4]
```diff
@@ -66,6 +66,10 @@
             {
                 "command": "kuskus.loadSymbols",
                 "title": "[Kuskus] Load Symbols from Cluster"
+            },
+            {
+                "command": "kuskus.loadTable",
```
Comment (rosshamish) [inline kusto-language-server/package.json:71]: Does this change fit with this PR? I don't recall seeing a `loadTable` command elsewhere.

**Comment by rosshamish** [comment]
> ok i pushed some slight changes to my current master - do i need to open another pull request?
> 
> basically i just made it so common is the value in tenant id, also just added a little info to the prompt that tenant id can be found in Azure Portal / Azure Active Directory
> 
> **edit: just realized on my package-lock.json i had updated version to 1.0.2 and author to my name so i could test a .vsix install**

1. It looks like you're good to update this PR by just pushing to master. I see your latest commit on this PR. I'm not very familiar with github's PR flow so I'm learning as we go here.

2. Yeah if you could update author back to rosshamish I'd appreciate that! Would be strange for the author to bounce around between versions. Totally fine to bump the version number though - could you bump it to 1.1.0 ?

**Comment by rosshamish** [comment]
One last request - could you add your name and github @handle to the Contributors section of README.md at the repo root?

File: kusto-language-server/package.json [hunk 5]
```diff
@@ -2,8 +2,8 @@
     "name": "kuskus-kusto-language-server",
     "displayName": "[Kuskus] Kusto Language Server",
     "description": "Language server for the Kusto query language",
-    "version": "1.0.1",
-    "publisher": "rosshamish",
+    "version": "1.0.2",
+    "publisher": "stevenhurwitt",
```
Comment (rosshamish) [inline kusto-language-server/package.json:6]: @stevenhurwitt - I'm still interested in merging this PR if you're willing! I think all that's left to do is reset the publisher name here back to `rosshamish`

```suggestion
    "publisher": "stevenhurwitt",
```

**Review by rosshamish — APPROVED** [review]

**Comment by rosshamish** [comment]
Hey @stevenhurwitt, I went ahead and landed this change (with a few additional changes) in #63. Thank you for the contribution! If you're still using Azure Data Explorer / Kusto, give it a shot. It'll be published to the extension store in the next hour or so.

**Comment by stevenhurwitt** [comment]
ok sweet, thanks man! I haven't been using Kusto as much, but maybe this will inspire me to get back into it.

**Pull request closed without merging**